diff --git a/libcef/browser/osr/render_widget_host_view_osr.cc b/libcef/browser/osr/render_widget_host_view_osr.cc
index 85e6202b9..5476553cd 100644
--- a/libcef/browser/osr/render_widget_host_view_osr.cc
+++ b/libcef/browser/osr/render_widget_host_view_osr.cc
@@ -247,6 +247,10 @@ CefRenderWidgetHostViewOSR::CefRenderWidgetHostViewOSR(
 
   auto context_factory = content::GetContextFactory();
 
+  bool gpu_compositing_disabled =
+    content::GpuDataManagerImpl::GetInstance()->IsGpuCompositingDisabled();
+
+  use_proxy_output_ = use_shared_texture_ || gpu_compositing_disabled;
   // Matching the attributes from RecyclableCompositorMac.
   compositor_ = std::make_unique<ui::Compositor>(
       context_factory->AllocateFrameSinkId(), context_factory,
@@ -257,6 +261,7 @@ CefRenderWidgetHostViewOSR::CefRenderWidgetHostViewOSR(
   compositor_->SetDelegate(this);
   compositor_->SetRootLayer(root_layer_.get());
   compositor_->AddChildFrameSink(GetFrameSinkId());
+  compositor_->SetBackgroundColor(background_color_);
 
   content::RenderWidgetHostImpl* render_widget_host_impl =
       content::RenderWidgetHostImpl::From(render_widget_host_);
@@ -379,7 +384,8 @@ void CefRenderWidgetHostViewOSR::ShowWithVisibility(
     return;
   }
 
-  if (!content::GpuDataManagerImpl::GetInstance()->IsGpuCompositingDisabled() &&
+  if (!use_shared_texture_ && 
+    !content::GpuDataManagerImpl::GetInstance()->IsGpuCompositingDisabled() &&
       !browser_impl_ &&
       (!parent_host_view_ || !parent_host_view_->browser_impl_)) {
     return;
@@ -1015,6 +1021,7 @@ void CefRenderWidgetHostViewOSR::OnFrameComplete(
   DCHECK_EQ(begin_frame_source_.source_id(), ack.frame_id.source_id);
   DCHECK_EQ(begin_frame_number_, ack.frame_id.sequence_number);
   begin_frame_pending_ = false;
+  had_frame_ = true; 
 }
 
 void CefRenderWidgetHostViewOSR::OnRenderFrameMetadataChangedAfterActivation(
@@ -1057,6 +1064,10 @@ void CefRenderWidgetHostViewOSR::OnRenderFrameMetadataChangedAfterActivation(
 
 std::unique_ptr<viz::HostDisplayClient>
 CefRenderWidgetHostViewOSR::CreateHostDisplayClient() {
+  if (had_frame_) {
+    begin_frame_pending_ = false;
+  }
+
   host_display_client_ =
       new CefHostDisplayClientOSR(this, gfx::kNullAcceleratedWidget);
   host_display_client_->SetActive(true);
@@ -1876,11 +1887,14 @@ void CefRenderWidgetHostViewOSR::RemoveGuestHostView(
 
 void CefRenderWidgetHostViewOSR::InvalidateInternal(
     const gfx::Rect& bounds_in_pixels) {
-  if (video_consumer_) {
-    video_consumer_->RequestRefreshFrame(bounds_in_pixels);
-  } else if (host_display_client_) {
-    OnPaint(bounds_in_pixels, host_display_client_->GetPixelSize(),
-            host_display_client_->GetPixelMemory());
+  if (!use_shared_texture_)
+  {
+    if (video_consumer_) {
+      video_consumer_->RequestRefreshFrame(bounds_in_pixels);
+    } else if (host_display_client_) {
+      OnPaint(bounds_in_pixels, host_display_client_->GetPixelSize(),
+              host_display_client_->GetPixelMemory());
+    }
   }
 }
 
diff --git a/libcef/browser/osr/render_widget_host_view_osr.h b/libcef/browser/osr/render_widget_host_view_osr.h
index 967ac0f80..74992382b 100644
--- a/libcef/browser/osr/render_widget_host_view_osr.h
+++ b/libcef/browser/osr/render_widget_host_view_osr.h
@@ -408,8 +408,10 @@ class CefRenderWidgetHostViewOSR
   viz::StubBeginFrameSource begin_frame_source_;
   uint64_t begin_frame_number_ = viz::BeginFrameArgs::kStartingFrameNumber;
   bool begin_frame_pending_ = false;
+  bool had_frame_ = false;
 
   bool use_shared_texture_ = false;
+  bool use_proxy_output_ = false;
   bool sync_frame_rate_ = false;
   bool external_begin_frame_enabled_ = false;
   bool needs_external_begin_frames_ = false;
