diff --git third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
index 57a5afa3c5901..5f7078f59882e 100644
--- third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
+++ third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.cc
@@ -6586,6 +6586,12 @@ void WebGLRenderingContextBase::TexParameter(GLenum target,
         }
         break;
       }
+
+      // CFX: add CFX_BIND_GAME_VIEW option to bind the game texture to glTexture
+      if ((is_float && paramf == 0x47d87380) || (!is_float && parami == 0x47d87380)) {
+        break;
+      }
+
       if ((is_float && paramf != GL_CLAMP_TO_EDGE &&
            paramf != GL_MIRRORED_REPEAT && paramf != GL_REPEAT) ||
           (!is_float && parami != GL_CLAMP_TO_EDGE &&
diff --git third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.idl third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.idl
index 9635e2c8ace14..153b9daec9963 100644
--- third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.idl
+++ third_party/blink/renderer/modules/webgl/webgl_rendering_context_base.idl
@@ -371,6 +371,8 @@ interface mixin WebGLRenderingContextBase {
     const GLenum REPEAT                         = 0x2901;
     const GLenum CLAMP_TO_EDGE                  = 0x812F;
     const GLenum MIRRORED_REPEAT                = 0x8370;
+    // CFX: CFX_BIND_GAME_VIEW
+    const GLenum CFX_BIND_GAME_VIEW             = 0x47d87380;
 
     /* Uniform Types */
     const GLenum FLOAT_VEC2                     = 0x8B50;
