diff --git cef/libcef/browser/browser_host_base.cc cef/libcef/browser/browser_host_base.cc
index d8f6b040d..2c147128c 100644
--- cef/libcef/browser/browser_host_base.cc
+++ cef/libcef/browser/browser_host_base.cc
@@ -1080,6 +1080,22 @@ void CefBrowserHostBase::SendMouseWheelEvent(const CefMouseEvent& event,
   }
 }
 
+// CFX: Introduce native OSR mouse wheel event
+void CefBrowserHostBase::SendMouseWheelEventNative(const void* msg) {
+  if (!CEF_CURRENTLY_ON_UIT()) {
+    CHROME_MSG m = *(CHROME_MSG*)msg;
+    CEF_POST_TASK(CEF_UIT, base::BindOnce([](scoped_refptr<CefBrowserHostBase> self, const CHROME_MSG& m) {
+        self->SendMouseWheelEventNative(&m);
+    }, scoped_refptr(this), m));
+    return;
+  }
+
+  if (platform_delegate_) {
+    platform_delegate_->SendMouseWheelEventNative(msg);
+  }
+}
+//
+
 bool CefBrowserHostBase::IsValid() {
   return browser_info_->IsValid();
 }
diff --git cef/libcef/browser/browser_host_base.h cef/libcef/browser/browser_host_base.h
index 069dbc21d..182fcf0db 100644
--- cef/libcef/browser/browser_host_base.h
+++ cef/libcef/browser/browser_host_base.h
@@ -259,6 +259,12 @@ class CefBrowserHostBase : public CefBrowserHost,
   void SendMouseWheelEvent(const CefMouseEvent& event,
                            int deltaX,
                            int deltaY) override;
+// CFX: introduce native OSR mouse wheel event
+#if defined(OS_WIN)
+  void SendMouseWheelEventNative(const void* msg) override;
+#endif
+//
+
   bool SendDevToolsMessage(const void* message, size_t message_size) override;
   int ExecuteDevToolsMethod(int message_id,
                             const CefString& method,
diff --git cef/libcef/browser/browser_platform_delegate.cc cef/libcef/browser/browser_platform_delegate.cc
index 167b2af62..fe9b40542 100644
--- cef/libcef/browser/browser_platform_delegate.cc
+++ cef/libcef/browser/browser_platform_delegate.cc
@@ -321,6 +321,10 @@ void CefBrowserPlatformDelegate::SendTouchEvent(const CefTouchEvent& event) {
   NOTIMPLEMENTED();
 }
 
+void CefBrowserPlatformDelegate::SendMouseWheelEventNative(const void* msg) {
+  NOTIMPLEMENTED();
+}
+
 void CefBrowserPlatformDelegate::SetFocus(bool setFocus) {}
 
 void CefBrowserPlatformDelegate::SendCaptureLostEvent() {
diff --git cef/libcef/browser/browser_platform_delegate.h cef/libcef/browser/browser_platform_delegate.h
index 698913fc8..444cfd054 100644
--- cef/libcef/browser/browser_platform_delegate.h
+++ cef/libcef/browser/browser_platform_delegate.h
@@ -235,6 +235,9 @@ class CefBrowserPlatformDelegate {
   virtual void SendMouseWheelEvent(const CefMouseEvent& event,
                                    int deltaX,
                                    int deltaY);
+// CFX: Introduce native OSR mouse wheel event
+  virtual void SendMouseWheelEventNative(const void* msg);
+//
   virtual void SendTouchEvent(const CefTouchEvent& event);
 
   // Send focus event. The browser's WebContents may be NULL when this method is
diff --git cef/libcef/browser/osr/browser_platform_delegate_osr.cc cef/libcef/browser/osr/browser_platform_delegate_osr.cc
index 44e49530f..013b9a212 100644
--- cef/libcef/browser/osr/browser_platform_delegate_osr.cc
+++ cef/libcef/browser/osr/browser_platform_delegate_osr.cc
@@ -20,6 +20,9 @@
 #include "ui/display/screen.h"
 #include "ui/events/base_event_utils.h"
 
+//CFX: OSR mouse input event
+#include "ui/events/blink/web_input_event.h"
+
 CefBrowserPlatformDelegateOsr::CefBrowserPlatformDelegateOsr(
     std::unique_ptr<CefBrowserPlatformDelegateNative> native_delegate,
     bool use_shared_texture,
@@ -163,6 +166,20 @@ void CefBrowserPlatformDelegateOsr::SendTouchEvent(const CefTouchEvent& event) {
   }
 }
 
+// CFX: Introduce native OSR mouse wheel event
+void CefBrowserPlatformDelegateOsr::SendMouseWheelEventNative(const void* msg) {
+  CefRenderWidgetHostViewOSR* view = GetOSRHostView();
+  if (!view) {
+    return;
+  }
+
+#if defined(OS_WIN)
+  blink::WebMouseWheelEvent web_event =
+      ui::MakeWebMouseWheelEvent(ui::MouseWheelEvent(*(CHROME_MSG*)msg));
+  view->SendMouseWheelEvent(web_event);
+#endif
+}
+
 void CefBrowserPlatformDelegateOsr::SetFocus(bool setFocus) {
   CefRenderWidgetHostViewOSR* view = GetOSRHostView();
   if (view) {
diff --git cef/libcef/browser/osr/browser_platform_delegate_osr.h cef/libcef/browser/osr/browser_platform_delegate_osr.h
index cdc84ed35..1d869a8ae 100644
--- cef/libcef/browser/osr/browser_platform_delegate_osr.h
+++ cef/libcef/browser/osr/browser_platform_delegate_osr.h
@@ -42,6 +42,9 @@ class CefBrowserPlatformDelegateOsr
   void SendMouseWheelEvent(const CefMouseEvent& event,
                            int deltaX,
                            int deltaY) override;
+// CFX: Introduce native OSR mouse wheel event
+  void SendMouseWheelEventNative(const void* msg) override;
+//
   void SendTouchEvent(const CefTouchEvent& event) override;
   void SetFocus(bool setFocus) override;
   gfx::Point GetScreenPoint(const gfx::Point& view,
