diff --git cef/libcef/common/chrome/chrome_main_delegate_cef.cc cef/libcef/common/chrome/chrome_main_delegate_cef.cc
index f733f0e0a..a7f4672f9 100644
--- cef/libcef/common/chrome/chrome_main_delegate_cef.cc
+++ cef/libcef/common/chrome/chrome_main_delegate_cef.cc
@@ -409,6 +409,7 @@ void ChromeMainDelegateCef::PreSandboxStartup() {
   }
   if (!resources_dir.empty()) {
     base::PathService::Override(chrome::DIR_RESOURCES, resources_dir);
+    base::PathService::Override(base::DIR_ASSETS, resources_dir);
   }
 
   if (command_line->HasSwitch(switches::kLocalesDirPath)) {
